### DISCUSSION TITLE
albums wont be recognized

### DISCUSSION THREAD

**Discussion by jannikmenzel** [MissingCore/Music#124] [Q&A] [answered]
Hi everyone, I have an issue according the creation of albums. My mp3 files are all sorted by adding an "album" as metadata. But somehow the music player wont recognize and sort them in the albums tab accordingly. I would appreciate your help.

**Comment by cyanChill** [comment, marked as answer]
> [!NOTE]
> As of [`v2.4.0-rc.1`](https://github.com/MissingCore/Music/releases/tag/v2.4.0-rc.1), albums are now unique by their album name & album artist.
>
> This means albums can now support tracks with different release years (ie: duplicate albums will no longer be created due to the tracks within them having different release years).
>
> In additional, local metadata editor support was added so you can fix these issues manually if they haven't been caught.

How albums are created in the app is through a unique combination of the following:
1. **[REQUIRED]** Album Name
2. **[REQUIRED]** Album Artist (falls back to the track artist if this isn't present)
3. Release Year

I'm not exactly sure what your album tab looks like, but here are some scenarios and causes that I can come up with in the top of my head:
- **No album for a set of tracks:** Both the album name and artist name has to be present. As you mention the "album" metadata is present, then it means that the artist name isn't present.
- **Multiple copies of the same albums with slightly different values:** This is caused from inconsistent metadata across your tracks such as:
  - Inconsistent spelling of the value used for the album name or artist name.
  - The album artist metadata not being present, which in that case, the artist of the track is used as a fallback, which if they're different across the different tracks in the album, it'll cause multiple albums with the same name but different artists to be created.
  - Inconsistent year values.
